<?xml version="1.0"?>
<launch>

    <arg name="config_file" default="$(find pybullet_ros)/ros/config/robots/franka_config.yaml"/>
    <!-- e.g. from my_ros_pkg.my_env import Environment ; "my_env" is what you set in environment param below-->
    <arg name="plugin_import_prefix" default="pybullet_ros.plugins"/>
    <arg name="environment" default="environment"/> <!-- name of python file without the .py inside plugins folder -->
    <arg name="pybullet_gui" default="True"/>
    <arg name="gui_options" default=""/>
    <arg name="rviz_bringup" default="True"/>
    <arg name="rviz_config" default="$(find pybullet_ros)/ros/config/rviz/franka.rviz"/>
    <arg name="start_paused" default="False"/>
    <arg name="parallel_plugin_execution" default="True"/>
    <arg name="use_deformable_world" default="False"/> <!-- enable/disable soft body simulation -->

    <group ns="franka">
        <arg name="urdf_path" default="$(find franka_panda_description)/robots/panda_arm.urdf.xacro"/>
        <param name="urdf_path" value="$(arg urdf_path)"/>
        <param name="pose_x" value="0.0"/>
        <param name="pose_y" value="0.0"/>
        <param name="pose_z" value="0.0"/>
        <param name="pose_yaw" value="0.0"/>
        <param name="fixed_base" value="True"/>
        <param name="use_inertia_from_file" value="True"/>

        <!-- upload urdf model to ROS param server -->
        <param name="robot_description" command="$(find xacro)/xacro $(arg urdf_path)"/>
<!--         <param name="robot_description" command="$(find xacro)/xacro $(arg urdf_path) arm_id:=papa xyz:='0 0 0'"/> -->

        <!-- robot_state_publisher, publish tf based on /joint_states topic and robot_description param -->
        <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen"/>
    </group>


    <!-- pybullet ros wrapper -->
    <include file="$(find pybullet_ros)/ros/launch/pybullet_ros.launch">
        <arg name="config_file" value="$(arg config_file)"/>
        <arg name="plugin_import_prefix" value="$(arg plugin_import_prefix)"/>
        <arg name="environment" value="$(arg environment)"/>
        <arg name="pybullet_gui" value="$(arg pybullet_gui)"/>
        <arg name="gui_options" value="$(arg gui_options)"/>
        <arg name="rviz_bringup" value="$(arg rviz_bringup)"/>
        <arg name="rviz_config" value="$(arg rviz_config)"/>
        <arg name="start_paused" value="$(arg start_paused)"/>
        <arg name="parallel_plugin_execution" value="$(arg parallel_plugin_execution)"/>
        <arg name="use_deformable_world" value="$(arg use_deformable_world)"/>
    </include>

</launch>
